add_libmagic.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+install_source() {
+    # install from source
+    # https://www.darwinsys.com/file/
+    # https://github.com/file/file/blob/FILE5_45/INSTALL#L51
+    (
+        version="file-5.45" &&
+        tmpfile="$(mktemp)" &&
+        curl -sSLo "${tmpfile}" "https://astron.com/pub/file/${version}.tar.gz" &&
```

Comment:
Tried it on ubuntu's Docker, fails here :)
`bash: curl: command not found`

Reply:
yes, the script is only intended for github actions default runner images and the cibuildwheel docker images (both have curl and make)